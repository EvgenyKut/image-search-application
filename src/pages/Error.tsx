import React from 'react';
import { useNavigate } from 'react-router-dom';
import { Container } from '../components';
import NotFound from '../assets/404.gif';
import { Routes } from '../constants/routes';

const Error = () => {
  const navigate = useNavigate();
  return (
    <Container>
      <img
        src={NotFound}
        alt="404..."
        onClick={() => navigate(Routes.HOME)}
        style={{ maxWidth: '100%', cursor: 'pointer' }}
      />
    </Container>
  );
};
export default Error;
