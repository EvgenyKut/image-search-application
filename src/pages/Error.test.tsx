import { fireEvent, render, screen } from '@testing-library/react';
import Error from './Error';
import { useNavigate } from 'react-router-dom';

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(() => jest.fn),
}));

describe('Error render test', () => {
  const navigate = jest.fn();
  const useNavigateMock = useNavigate;
  beforeEach(() => {
    (useNavigateMock as jest.Mock).mockImplementation(() => navigate);
  });
  afterEach(() => {
    (useNavigateMock as jest.Mock).mockClear();
  });

  it('Error test case', () => {
    render(<Error />);
    const errorImage = screen.getByRole('img');
    expect(errorImage).toBeInTheDocument();
    expect(errorImage).toBeVisible();
    fireEvent.click(errorImage);
    expect(navigate).toHaveBeenLastCalledWith('/');
    expect(errorImage).toHaveStyle({ cursor: 'pointer' });
  });
});
