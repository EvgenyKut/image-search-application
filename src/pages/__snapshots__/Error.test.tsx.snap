// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error render test snapshot 1`] = `
<div
  className="container"
  data-testid="container"
>
  <img
    alt="404..."
    onClick={[Function]}
    src="404.gif"
    style={
      Object {
        "cursor": "pointer",
        "maxWidth": "100%",
      }
    }
  />
</div>
`;
